File: cpp/649_Dota2_参议院_medium.cpp
class Solution {
public:
    string predictPartyVictory(string senate) {
        int N = senate.length();
        queue<int> radiant, dire;
        for(int i=0; i<N; i++)
            if(senate[i]=='R')
                radiant.push(i);
            else
                dire.push(i);
        while(!radiant.empty() && !dire.empty()){
            if(radiant.front() < dire.front())
                radiant.push(radiant.front()+N);
            else
                dire.push(dire.front()+N);
            radiant.pop();
            dire.pop();
        }
        return radiant.empty() ? "Dire": "Radiant";
    }
};

File: python/649_Dota2_参议院_medium.py
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        N = len(senate)
        radiant = collections.deque()
        dire = collections.deque()
        for index, ch in enumerate(senate):
            if ch == 'R':
                radiant.append(index)
            else:
                dire.append(index)
        while radiant and dire:
            if radiant[0] < dire[0]:
                radiant.append(radiant[0]+N)
            else:
                dire.append(dire[0]+N)
            radiant.popleft()
            dire.popleft()
        return 'Radiant' if radiant else 'Dire'